controllers/dynakube/updates/version.go

```diff
@@ -104,17 +125,15 @@ func updateImageVersion(
 		}
 	}
 
-	rec.Log.Info("Update found",
-		"image", img,
-		"oldVersion", target.Version, "newVersion", ver.Version,
-		"oldHash", target.ImageHash, "newHash", ver.Hash)
-
+	logMessage := fmt.Sprintf("Update found image: %s, oldVersion: %s, newVersion: %s, oldHash: %s, newHash: %s", img, target.Version, ver.Version, target.ImageHash, ver.Hash)
+	rec.Log.Info(logMessage)
+	recorder.Event(dk, corev1.EventTypeNormal, UpdateImageVersionEvent, logMessage)
```

Comment:
if the event is fired after the method exits (if possible), the instance would have to be passed to it.